src/plugins/layout.ts

```diff
@@ -73,201 +68,153 @@ export default class Layout implements IPlugin {
     }
   }
 
-  // Recording full layouts of all elements on the page at once is an expensive operation
-  // and can impact user's experience by hanging the page due to occupying the thread for too long
-  // To avoid this, we only assign indices to all elements and build a ShadowDom with dummy layouts
-  // just to have a valid DOM skeleton. After that, we can come back to dummy layouts and populate
-  // them with real data asynchronously (if it takes too long to do at once) by yielding a thread
-  // and returning to it later through a set timeout
   private discoverDom() {
```

Comment:
>discoverDom [](start = 10, length = 11)

I believe all these changes are to handle discover phase different from rest of the updates. I think we can avoid that and keep the scope of the changes small. We can discuss more offline and finalize this decision.